packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -0,0 +1,182 @@
+};
+
+const defaultProps = {
+  isEndDateRange: false,
+  isStartDateRange: false,
+  withPortal: undefined,
+};
+
+class DatePicker extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: props.startDate,
+      endDate: props.endDate,
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(date) {
+    this.setState({
+      startDate: date,
+    });
+
+    if (this.props.onChange) {
+      this.props.onChange(date);
+    }
+  }
+
+  render() {
+    const {
+      endDate,
+      excludeDates,
+      filterDate,
+      includeDates,
+      maxDate,
+      minDate,
+      isEndDateRange,
+      isStartDateRange,
+      selectedDate,
+      startDate,
+      withPortal,
+      ...customProps
+    } = this.props;
+    const classes = classNames(['terra-DatePicker']);
+
+    // TODO: Need translation from date_util
+    const todayString = 'Today';
+
+    // TODO: Get the locale from date_util
+    const userLocale = window.navigator.language;
+
+    // TODO: Get date format from date_util
+    const localMoment = moment();
+    localMoment.locale(userLocale);
+    const momentDateFormat = localMoment.localeData().longDateFormat('L');
+
+    const portalPicker =
+      (<ReactDatePicker
+        {...customProps}
+        selected={selectedDate || this.state.startDate}
+        onChange={this.handleChange}
+        customInput={<DateInput />}
+        endDate={endDate}
+        excludeDates={excludeDates}
+        filterDate={filterDate}
+        includeDates={includeDates}
+        maxDate={maxDate}
+        minDate={minDate}
+        selectsEnd={isEndDateRange}
+        selectsStart={isStartDateRange}
+        startDate={startDate}
+        todayButton={todayString}
+        withPortal={withPortal === undefined ? true : withPortal}
+        dateFormatCalendar=" "
+        dateFormat={momentDateFormat}
+        fixedHeight
+        locale={userLocale}
+        placeholderText={momentDateFormat}
+        dropdownMode={'select'}
+        showMonthDropdown
+        showYearDropdown
+      />);
+
+    const popupPicker =
+      (<ReactDatePicker
+        {...customProps}
+        selected={selectedDate || this.state.startDate}
+        onChange={this.handleChange}
+        customInput={<DateInput />}
+        endDate={endDate}
+        excludeDates={excludeDates}
+        filterDate={filterDate}
+        includeDates={includeDates}
+        maxDate={maxDate}
+        minDate={minDate}
+        selectsEnd={isEndDateRange}
+        selectsStart={isStartDateRange}
+        startDate={startDate}
+        todayButton={todayString}
+        withPortal={withPortal === undefined ? false : withPortal}
+        dateFormatCalendar=" "
+        dateFormat={momentDateFormat}
+        fixedHeight
+        locale={userLocale}
+        placeholderText={momentDateFormat}
+        dropdownMode={'select'}
+        showMonthDropdown
+        showYearDropdown
+      />);
```

Comment:
I'm a little on the fence if this is how the ResponsiveElement should be used, I think for right now this is what we have to work with, but this seems like a case where the same element just needs to be rendered differently, rather than swapping out an entire element for another. It looks like only one parameter is different between these huge blocks.

I'm wondering if it would be possible to wire in a onChange event that can be passed into the ResponsiveElement or maybe a new component that is called back with the current breakpoint size so an Element can render itself differently.